### DISCUSSION TITLE
Why fastAI?

### DISCUSSION THREAD

**Discussion by Aniruddha120** [mrdbourke/pytorch-deep-learning#440] [Q&A]
Dear All,
Hope you are doing well. In the learnpytorch.io book, Daneil has recommended to follow pytorch and fastai(at extra resources)(pls look at the ss also).
But if I am learning pytorch and can do almost everything, why another framework? can you guys enlighten me?
![image](https://user-images.githubusercontent.com/34673269/236636742-5be05939-78ef-4e25-b469-f52dfe7e2b48.png)

**Comment by mrdbourke** [comment]
Hi @Aniruddha120,

You're right, PyTorch is very capable of doing many things.

However, frameworks such as fast.ai build on top of PyTorch and implement several helpful utilities (that you could of course do with base PyTorch) without having to rewrite a large amount of code.

For example, PyTorch training and testing loops can be a bit cumbersome to fully write.

Where as fast.ai and others implement training/testing classes that have a fair bit of functionality out of the box.

If you've ever done web development it's similar to using pure JavaScript (PyTorch) vs React (fast.ai).

In short: 
- You want to implement everything from scratch: use PyTorch
- You want to just get models training and working: use fast.ai/other frameworks like Lightning (runs PyTorch underneath)
